File: src/main/java/no/nav/syfo/domain/InngaaendeJournal.java
package no.nav.syfo.domain;

import lombok.Builder;
import lombok.Getter;
import lombok.Value;

@Value
@Getter
@Builder
public class InngaaendeJournal {
    public static final String MIDLERTIDIG = "MIDLERTIDIG";

    String dokumentId;
    String status;
}

File: src/main/kotlin/no/nav/syfo/domain/InngaaendeJournal.kt
package no.nav.syfo.domain

import java.time.LocalDateTime

data class InngaaendeJournal(
    val dokumentId: String,
    val status: JournalStatus,
    val mottattDato: LocalDateTime,
)

/**
 * https://confluence.adeo.no/display/BOA/saf+-+Enums
 */
enum class JournalStatus {
    MOTTATT, // Tidligere: MIDLERTIDIG
    UKJENT, // Tidligere: ANNET
    FERDIGSTILT, // Tidligere: ENDELIG
    MIDLERTIDIG,
}
